inc/class-core-sitemaps.php

```diff
@@ -232,4 +233,24 @@ public function render_sitemaps() {
 			exit;
 		}
 	}
+
+	/**
+	 * Redirect an URL to the wp-sitemap.xml
+	 *
+	 * @param bool     $bypass Pass-through of the pre_handle_404 filter value.
+	 * @param WP_Query $query The WP_Query object.
+	 */
+	public function redirect_sitemapxml( $bypass, $query ) {
+		// If a plugin has already utilized the pre_handle_404 function, return without action to avoid conflicts.
+		if ( $bypass ) {
+			return $bypass;
+		}
+
+		// 'pagename' is for most permalink types, name is for when the %postname% is used as a top-level field.
+		if ( 'sitemap-xml' === $query->get( 'pagename' ) ||
```

Comment:
what about a more generic check with `preg_match` to search everything that has something `sitemap` in the name?

Reply:
I think we should leave this as-is for now to be within scope of #144.

A catch-all for anything with `sitemap` in it sounds very dangerous and could cause unexpected behavior.

We'd want to catch the case where people try to access `/sitemap.xml`, as that's by far the most common name.